frontend/src/hocs/withHostCommunication/withHostCommunication.tsx

```diff
@@ -87,7 +88,7 @@ function withHostCommunication(
         if (
           !origin ||
           message.stCommVersion !== HOST_COMM_VERSION ||
-          !CLOUD_COMM_WHITELIST.find(el => isValidURL(el, origin))
+          !allowedOrigins.find(el => isValidURL(el, origin))
```

Comment:
The `isValidURL` function expects a hostname, but if the origin fails to parse for some reason, the origin itself is used instead of the hostname, meaning that if an attacker could construct some origin that would match the pattern of the hostname but which would not be parsed as a valid URL, they would still be able to access the interface.

Additionally, only the hostname if validated, so a page on a different protocol or port could still interact with the service (for example a mitm'd HTTP connection or load balancer misconfiguration )

Reply:
Hm, I wasn't able to think of a scenario under normal usage where the event origin would fail to parse into a valid URL but would be useful for us to attempt to match against anyway, so I'm fairly certain we can get around this issue by just considering the match to have failed when the origin fails to parse.

I also changed `isValidUrl` into `isValidOrigin` and made it more strict about protocol/port checks to handle those issues.

Reply:
Hm, I wasn't able to think of a scenario under normal usage where the event origin would fail to parse into a valid URL but would be useful for us to attempt to match against anyway, so I'm fairly certain we can get around this issue by just considering the match to have failed when the origin fails to parse.

I also changed `isValidUrl` into `isValidOrigin` and made it more strict about protocol/port checks to handle those issues.